app/services/provider_interface/hesa_data_export.rb

```diff
@@ -1,22 +1,26 @@
 module ProviderInterface
   class HesaDataExport
-    def initialize(provider_ids:)
-      @provider_ids = provider_ids
+    def initialize(actor:)
+      @provider_ids = actor.providers.map(&:id)
+      @auth = ProviderAuthorisation.new(actor: actor)
     end
 
     def call
-      applications = ApplicationChoice
+      applications_join = ApplicationChoice
         .includes(
           :course,
           :provider,
           :site,
           course_option: { course: %i[provider accredited_provider] },
           application_form: %i[candidate application_qualifications],
-        ).where(
+        )
+
+      applications =
+        applications_join.where('providers.id': @provider_ids)
+        .or(applications_join.where('courses_course_options.accredited_provider_id': @provider_ids))
```

Comment:
Could we use `GetApplicationChoicesForProviders` here? We could pass in the `includes` if necessary